src/frontends/jax/src/utils.hpp

```diff
@@ -0,0 +1,216 @@
+
+const std::string& get_jax_prefix();
+
+/// \brief Macro to check whether a boolean condition holds.
+/// \param COND Condition to check
+/// \param ... Additional error message info to be added to the error message via the `<<`
+///            stream-insertion operator. Note that the expressions here will be evaluated lazily,
+///            i.e., only if the `cond` evalutes to `false`.
+/// \throws ::ov::frontend::OpConversionFailure if `cond` is false.
+#ifndef JAX_OP_CONVERSION_CHECK
+#    define JAX_OP_CONVERSION_CHECK(COND, ...) \
+        OPENVINO_ASSERT_HELPER(::ov::frontend::OpConversionFailure, "", (COND), get_jax_prefix(), __VA_ARGS__)
+#endif
+
+void num_inputs_check(const NodeContext& context, size_t min_inputs, size_t max_inputs);
+
+Output<Node> make_optional_bias(const Output<Node>& base_op,
+                                const NodeContext& context,
+                                int bias_input_idx,
+                                const std::vector<int>& unsqueeze_dims = {});
+
+Output<Node> reshape_channelwise(const NodeContext& context,
+                                 const Output<Node>& data,
+                                 const Output<Node>& shape_source);
+
+std::tuple<Output<Node>, Output<Node>> get_shape_rank(const NodeContext& context,
+                                                      const Output<Node>& x,
+                                                      bool as_scalar = false,
+                                                      element::Type output_type = element::i32);
+
+Output<Node> reshape_kernel_for_group(const NodeContext& context, const Output<Node>& kernel, int64_t groups);
+
+std::shared_ptr<Node> get_axes_range(const NodeContext& context, int input_id);
+
+std::shared_ptr<Node> get_node_axes_range(const NodeContext& context, const Output<Node>& x);
+
+Output<Node> normalize_axis(const NodeContext& context, const Output<Node>& axis, const Output<Node>& input_node);
+
+std::shared_ptr<Node> numel(const NodeContext& context,
+                            const Output<Node>& x,
+                            element::Type output_type = element::i32);
+
+element::Type convert_dtype(int64_t dtype_value);
+
+Output<Node> apply_dtype(const NodeContext& context, size_t dtype_port, const Output<Node>& input_tensor);
+
+op::PadType convert_pad(const std::string& pt_pad);
+
+Output<Node> concat_list_construct(const Output<Node>& input);
+
+/// \brief Checks if input represents empty list.
+/// \param input Input to check.
+/// \return true if input is empty list, false - if input is non-empty or non-list.
+bool is_empty_list(const Output<Node>& input);
+
+OutputVector make_framework_node_ignore_bodies(const NodeContext& context, const std::string& exception);
+OutputVector make_framework_node(const NodeContext& context, const std::string& exception);
+
+std::shared_ptr<op::util::FrameworkNode> cast_fw_node(std::shared_ptr<Node> node, const std::string& type);
+
```

Comment:
please remove unneeded routines

Reply:
The unneeded routines have been removed.